test/gtest/uct/test_event.cc

```diff
@@ -109,6 +110,9 @@ void test_uct_event_fd::test_recv_am(bool signaled)
     recv_buffer = (recv_desc_t *) malloc(sizeof(*recv_buffer) + sizeof(send_data));
     recv_buffer->length = 0; /* Initialize length to 0 */
 
+    /* give a chance to finish connection for some transports (ib/ud, tcp) */
+    UCT_TEST_WAIT_RESOURCE_READY(m_e1->ep(0));
```

Comment:
maybe just use existing  uct_test::flush? It flushes ifaces, but would work here, I guess.
It's advantage is that it performs flush for a certain deadline (which is quite small ~ 10 secs) to avoid hangs. 
Or you can implement similar ep flush as uct_test member rather than macro

Reply:
thanks! done